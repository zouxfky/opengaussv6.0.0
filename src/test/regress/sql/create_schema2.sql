declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;

    query_str := 'create schema "' || var_name || '"';
    EXECUTE IMMEDIATE query_str;
    
    query_str := 'drop schema "' || var_name ||'" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/

declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;

    query_str := 'create schema authorization "' || var_name || '"';
    EXECUTE IMMEDIATE query_str;
    
    query_str := 'drop schema "' || var_name ||'" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/

declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;
    
    query_str := 'create schema "' || var_name ||'_123"';
    EXECUTE IMMEDIATE query_str;

    query_str := 'drop schema "' || var_name || '_123" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/

create schema gs_role;
create schema gs_role_;
create schema gs_role_abc;

drop schema gs_role;
drop schema gs_role_;
drop schema gs_role_abc;