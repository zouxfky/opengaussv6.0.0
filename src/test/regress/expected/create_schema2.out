declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;

    query_str := 'create schema "' || var_name || '"';
    EXECUTE IMMEDIATE query_str;
    
    query_str := 'drop schema "' || var_name ||'" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/
--?.*
DETAIL:  schema name can not same as initial account
--?.*
PL/pgSQL function inline_code_block line 7 at EXECUTE statement
declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;

    query_str := 'create schema authorization "' || var_name || '"';
    EXECUTE IMMEDIATE query_str;
    
    query_str := 'drop schema "' || var_name ||'" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/
--?.*
DETAIL:  schema name can not same as initial account
--?.*
PL/pgSQL function inline_code_block line 7 at EXECUTE statement
declare
    var_name text;
    query_str text;
begin
    select rolname into var_name from pg_authid where oid=10;
    
    query_str := 'create schema "' || var_name ||'_123"';
    EXECUTE IMMEDIATE query_str;

    query_str := 'drop schema "' || var_name || '_123" CASCADE';
    EXECUTE IMMEDIATE query_str;
end;
/
create schema gs_role;
create schema gs_role_;
ERROR:  unacceptable schema name "gs_role_"
DETAIL:  The prefix "gs_role_" is reserved.
create schema gs_role_abc;
ERROR:  unacceptable schema name "gs_role_abc"
DETAIL:  The prefix "gs_role_" is reserved.
drop schema gs_role;
drop schema gs_role_;
ERROR:  schema "gs_role_" does not exist
drop schema gs_role_abc;
ERROR:  schema "gs_role_abc" does not exist
